source/dol/bases/d_mj2d_data.cpp

```diff
@@ -0,0 +1,312 @@
+    if (mStaffRollHighScore < score ) {
+        mStaffRollHighScore = score;
+    }
+}
+
+int dMj2dGame_c::getStaffCreditHighScore() const {
+    return mStaffRollHighScore;
+}
+
+void dMj2dGame_c::setHintMovieBought(int movie) {
+    mHintMovieBought[movie] = true;
+}
+
+bool dMj2dGame_c::getHintMovieBought(int movie) const {
+    return mHintMovieBought[movie] != 0;
+}
+
+void dMj2dGame_c::setCollectCoin(int world, int level, u8 coins) {
+    onCourseDataFlag(world, level, coins & COIN_MASK);
+}
+
+int dMj2dGame_c::getTotalWorldCollectCoin(int world) {
+    int coinCount = 0;
+    for (int level = 0; level < STAGE_COUNT; level++) {
+        for (u32 coin = 0; coin < STAR_COIN_COUNT; coin++) {
+            if (isCollectCoin(world, level, coin) & 0xFF) {
+                coinCount++;
+            }
+        }
+    }
+
+    return coinCount;
+}
+
+u8 dMj2dGame_c::isCollectCoin(int world, int level, int coin) const {
+    return 1 << coin & mStageCompletion[world][level];
+}
+
+void dMj2dGame_c::setStartKinokoKind(int world, u8 type) {
+    mStartKinokoType[world] = type;
+}
+
+int dMj2dGame_c::getStartKinokoKind(int world) const {
+    return mStartKinokoType[world];
+}
+
+void dMj2dGame_c::setDeathCount(int world, int level, bool isSwitchPressed, u8 count) {
+    if (count >= SUPER_GUIDE_DEATH_COUNT && world < NORMAL_WORLD_COUNT && (level <= STAGE_CASTLE || level == STAGE_DOOMSHIP)) {
+        mGameCompletion |= SUPER_GUIDE_TRIGGERED;
+    }
+
+    if (isSwitchPressed && world == WORLD_3 && level == STAGE_4) {
+        setSwitchDeathCount(count);
+    } else {
+        mDeathCount[world][level] = count;    
+    }
+}
+
+int dMj2dGame_c::getDeathCount(int world, int level, bool isSwitchPressed) const {
+    if (isSwitchPressed && world == WORLD_3 && level == STAGE_4) {
+        return getSwitchDeathCount();
+    }
+
+    return mDeathCount[world][level];
+}
+
+void dMj2dGame_c::setSwitchDeathCount(u8 count) {
+    mDeathCountSwitch = count;
+}
+
+int dMj2dGame_c::getSwitchDeathCount() const {
+    return mDeathCountSwitch;
+}
+
+void dMj2dGame_c::setContinue(int player, s8 continues) {
+    mPlayerContinue[player] = continues;
+}
+
+s8 dMj2dGame_c::getContinue(int player) const {
+    return mPlayerContinue[player];
+}
+void dMj2dGame_c::setStockItem(int item, s8 count) {
+    if (count > MAX_STOCK_ITEM) {
+        count = MAX_STOCK_ITEM;
+    }
+
+    mStockItemCount[item] = count;
+}
+
+u8 dMj2dGame_c::isWorldDataFlag(int world, u8 flag) const {
+    return flag & mWorldCompletion[world];
+}
+
+void dMj2dGame_c::onWorldDataFlag(int world, u8 flag) {
+    mWorldCompletion[world] |= flag;
+}
+
+void dMj2dGame_c::offWorldDataFlag(int world, u8 flag) {
+    mWorldCompletion[world] &= ~flag;
+}
+
+int dMj2dGame_c::getCourseDataFlag(int world, int level) const {
+    return mStageCompletion[world][level];
+}
+
+bool dMj2dGame_c::isCourseDataFlag(int world, int level, ulong flag) const {
+    return (flag & mStageCompletion[world][level]) != 0;
```

Comment:
```suggestion
    return (mStageCompletion[world][level] & flag) != 0;
```
would be a bit more readable imo (ignore this if it wouldn't match anymore ofc)